spatialdata/_core/_rasterize.py

```diff
@@ -0,0 +1,383 @@
+        elements = getattr(sdata, element_type)
+        for name, element in elements:
+            rasterized = rasterize(
+                data=element,
+                axes=axes,
+                min_coordinate=min_coordinate,
+                max_coordinate=max_coordinate,
+                target_coordinate_system=target_coordinate_system,
+                target_unit_to_pixels=target_unit_to_pixels,
+                target_width=target_width,
+                target_height=target_height,
+                target_depth=target_depth,
+            )
+            new_name = f"{name}_rasterized_{element_type}"
+            new_images[new_name] = rasterized
+    return SpatialData(images=new_images, table=sdata.table)
+
+
+@rasterize.register(SpatialImage)
+@rasterize.register(MultiscaleSpatialImage)
+def _(
+    data: SpatialImage,
+    axes: tuple[str, ...],
+    min_coordinate: ArrayLike,
+    max_coordinate: ArrayLike,
+    target_coordinate_system: str,
+    target_unit_to_pixels: Optional[float] = None,
+    target_width: Optional[float] = None,
+    target_height: Optional[float] = None,
+    target_depth: Optional[float] = None,
+) -> SpatialImage:
+    # get dimensions of the target image
+    spatial_axes = get_spatial_axes(axes)
+    target_width, target_height, target_depth = _compute_target_dimensions(
+        spatial_axes=spatial_axes,
+        min_coordinate=min_coordinate,
+        max_coordinate=max_coordinate,
+        target_unit_to_pixels=target_unit_to_pixels,
+        target_width=target_width,
+        target_height=target_height,
+        target_depth=target_depth,
+    )
+    target_sizes = {
+        "x": target_width,
+        "y": target_height,
+        "z": target_depth,
+    }
+
+    # get inverse transformation
+    transformation = get_transformation(data, target_coordinate_system)
+    dims = get_dims(data)
+    assert isinstance(transformation, BaseTransformation)
+    affine = _get_affine_for_element(data, transformation)
+    target_axes_unordered = affine.output_axes
+    assert set(target_axes_unordered) in [{"x", "y", "z"}, {"x", "y"}, {"c", "x", "y", "z"}, {"c", "x", "y"}]
+    target_axes: tuple[str, ...]
+    if "z" in target_axes_unordered:
+        if "c" in target_axes_unordered:
+            target_axes = ("c", "z", "y", "x")
+        else:
+            target_axes = ("z", "y", "x")
+    else:
+        if "c" in target_axes_unordered:
+            target_axes = ("c", "y", "x")
+        else:
+            target_axes = ("y", "x")
+    target_spatial_axes = get_spatial_axes(target_axes)
+    assert len(target_spatial_axes) == len(min_coordinate)
+    assert len(target_spatial_axes) == len(max_coordinate)
+    corrected_affine = affine.to_affine(input_axes=axes, output_axes=target_spatial_axes)
+
+    # get xdata
+    if isinstance(data, SpatialImage):
+        xdata = data
+        pyramid_scale = None
+    elif isinstance(data, MultiscaleSpatialImage):
```

Comment:
I think this would be more clear to separate `SpatialImage` and `MultiscaleSpatialImage` into different single dispatch functions.

You can wrap the things in lines 260 through 334 in a separate function that is called by both the `SpatialImage` and `MultiscaleSpatialImage` single dispatch functions.

Reply:
I thought about it and I could 
* either go as you said for one single dispatch for images and one for multiscales, and have two functions wrapping the code to be reused: one until line 260, and one from line 334.
* either wrap the lines after "# get xdata" in a separate function and keep the rest in a unique single dispatch function. 

I prefer the second because execpt for the "# get xdata" part, the implementation is identical for single scale and multiscale (also is much more different regarding images vs labels than `SpatialImage` vs `MultiscaleSpatialImage`). I'll implement this.

Reply:
done.